include/gridtools/sid/composite.hpp

```diff
@@ -152,17 +154,20 @@ namespace gridtools {
                 struct composite_ptr {
                     static_assert(sizeof...(Keys) == sizeof...(Ptrs), GT_INTERNAL_ERROR);
 
-                    typename hymap::keys<Keys...>::template values<Ptrs...> m_vals;
+                    tuple<Ptrs...> m_vals;
                     GT_TUPLE_UTIL_FORWARD_GETTER_TO_MEMBER(composite_ptr, m_vals);
                     GT_TUPLE_UTIL_FORWARD_CTORS_TO_MEMBER(composite_ptr, m_vals);
                     GT_CONSTEXPR GT_FUNCTION decltype(auto) operator*() const {
-                        return tuple_util::host_device::transform([](auto const &ptr)
+                        return tuple_util::host_device::convert_to<hymap::keys<Keys...>::template values>(
+                            tuple_util::host_device::transform([](auto const &ptr)
 // Workaround for GCC 9 bug https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90333
-#if defined(__clang__) || (__GNUC__ != 9 && __GNUC_MINOR__ > 2)
-                                                                      GT_FORCE_INLINE_LAMBDA
+// The failure is observed with 9.3 as well even though they say it was already fixed there.
+// gcc 10.1, 10.2 fails here as well. Disabling for all gcc 9 and 10 versions...
+#if defined(__clang__) || !defined(__GNUC__) || (__GNUC__ != 9 && __GNUC__ != 10)
```

Comment:
```suggestion
#if !defined(__GNUC__) || (__GNUC__ != 9 && __GNUC__ != 10)
```
I think the clang check is useless.

Reply:
Nope.
Clang defines __GNUC__ / __GNUC_MINOR__ / __GNUC_PATCHLEVEL__ according to the version of gcc that it claims full compatibility with.

Reply:
Ah thanks!